src/main/java/com/puppycrawl/tools/checkstyle/checks/indentation/NewHandler.java

```diff
@@ -43,26 +48,57 @@ public NewHandler(IndentationCheck indentCheck,
 
     @Override
     public void checkIndentation() {
-        final DetailAST type = getMainAst().getFirstChild();
-        if (type != null) {
-            checkExpressionSubtree(type, getIndent(), false, false);
+        final DetailAST firstChild = getMainAst().getFirstChild();
+        final DetailAST mainAst = getMainAst();
+
+        // if new is on the line start and it is not the part of assignment.
+        if (isOnStartOfLine(mainAst)
+                && !getMainAst().getParent().getParent().getText().equals(ASSIGNMENT)) {
```

Comment:
`getParent().getParent().getText().equals(ASSIGNMENT)`
why dont check its type? Like `getMainAst().getParent().getParent().getType() == TokenTypes.ASSIGN`?

Reply:
Earlier It was giving error, when I compared. Type was showing VARIABLE_DEF, but now its all correct. I don't know how, Maybe I did some there. Now its fine. Thank you very much.

Reply:
New diff report: https://abhishek-kumar09.github.io/checkstyleDiffReportTempContribition/